runtime/frequency/src/lib.rs

```diff
@@ -150,6 +148,44 @@ impl Contains<RuntimeCall> for BaseCallFilter {
 	}
 }
 
+impl BaseCallFilter {
+	fn is_utility_call_allowed(call: &RuntimeCall) -> bool {
+		match call {
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::batch_all { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::force_batch { calls, .. }) =>
+				calls.iter().any(Self::is_batch_call_allowed),
+			_ => false,
+		}
+	}
+
+	fn is_batch_call_allowed(call: &RuntimeCall) -> bool {
+		match call {
+			// Block all nested `batch` calls from utility batch
+			RuntimeCall::Utility(pallet_utility::Call::batch { .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::batch_all { .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::force_batch { .. }) => false,
+
+			// Block all `FrequencyTxPayment` calls from utility batch
+			RuntimeCall::FrequencyTxPayment(..) => false,
+
+			// Block `create_provider` and `create_schema` calls from utility batch
+			RuntimeCall::Msa(pallet_msa::Call::create_provider { .. }) |
+			RuntimeCall::Schemas(pallet_schemas::Call::create_schema { .. }) => false,
+
+			// Block all `Pays::No` calls from utility batch
+			_ if Self::is_pays_no_call(call) => false,
```

Comment:
I do not think this is blocking // Block all `Pays::No` calls from the utility batch.